cmd/kubeadm/app/kubeadm.go

```diff
@@ -30,7 +30,8 @@ import (
 
 // Run creates and executes new kubeadm command
 func Run() error {
-	klog.InitFlags(nil)
+	local := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	klog.InitFlags(local)
```

Comment:
could you explain why this change is needed?

Reply:
Because the code moves and imports `k8s.io/component-base/log` package, the package has the init() function, which contains init log flag：
https://github.com/kubernetes/kubernetes/blob/a3806d6e497f5e235df9ae9157f29414a62dd3b4/staging/src/k8s.io/component-base/logs/logs.go#L34-L36

Leaving this line of code unchanged leads to `flag refined`

or do you have any better suggestions ? @neolit123

Reply:
> Because the log code change and kubeadm run imports k8s.io/component-base/log package

i cannot find direct imports of `k8s.io/component-base/log` in `/cmd/kubeadm`. does it get imported indirectly?

> Leaving this line of code klog.InitFlags(nil) unchanged leads to flag refined error

but this only happens after this PR, correct?

> the package has the init() function, which contains init log flag：

having something like:
```
func init() { 
 	klog.InitFlags(flag.CommandLine) 
 }
```

doesn't seem like a good idea in general.
this makes klog flags to be force added to the default command line for anyone who importers the `k8s.io/component-base/log` package even if they only want its utilities.

Reply:
if something else is doing the `klog.InitFlags` can we remove the `klog.InitFlags` call in kubeadm completely and replace it with a comment in the lines of:

> // package k8s.io/component-base/log is responsible for doing klog.InitFlags()

and add a `_ k8s.io/component-base/log` import on top?

Reply:
> if something else is doing the `klog.InitFlags` can we remove the `klog.InitFlags` call in kubeadm completely and replace it with a comment in the lines of:
> 
> > // package k8s.io/component-base/log is responsible for doing klog.InitFlags()
> 
> and add a `_ "k8s.io/component-base/log"` import on top?

Thanks, It looks better and clearer and It's easier to understand

Reply:
> i cannot find direct imports of k8s.io/component-base/log in /cmd/kubeadm. does it get imported indirectly?

it is indirect, and the following is the compilation package order：

```
package k8s.io/kubernetes/cmd/kubeadm/app
	imports k8s.io/kubernetes/cmd/kubeadm/app/cmd
	imports k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/validation
	imports k8s.io/kubernetes/cmd/kubeadm/app/componentconfigs
	imports k8s.io/kube-proxy/config/v1alpha1
	imports k8s.io/component-base/config/v1alpha1
	imports k8s.io/component-base/config/options
	imports k8s.io/component-base/logs
```
> doesn't seem like a good idea in general.
this makes klog flags to be force added to the default command line for anyone who imports the k8s.io/component-base/log package even if they only want its utilities.

so I use change following code instead of klog.InitFlags(nil) to avoid using logs init()
```
	local := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
	klog.InitFlags(local)
```
> but this only happens after this PR, correct?

This PR changes the following code and the following code uses k8s.io/component-base/config/options package,
https://github.com/kubernetes/kubernetes/pull/99693/files#diff-c0a54d42bf7c11099aaec980abd8ee871dfba9428da8b3ce8b8d480ef4d2989eR110-R113

k8s.io/component-base/config/options package import k8s.io/component-base/logs and causesk8s.io/component-base/logs init() function to be called

Reply:
> having something like:
> 
> ```
> func init() { 
>  	klog.InitFlags(flag.CommandLine) 
>  }
> ```
> 
> doesn't seem like a good idea in general.
> this makes klog flags to be force added to the default command line for anyone who imports the `k8s.io/component-base/log` package even if they only want its utilities.

Strongly agree with this. packages that have side effects when you import them should not contain other things that would give people reasons to import them. As an example, the `install` packages for our API types (e.g. `pkg/apis/core/install/install.go`) are standalone packages, rather than installing in the package that defines the API type.

Reply:
I agree with you. 
I think it's better to delete this log package init() function